src/htmx.js

```diff
@@ -1123,10 +1123,16 @@ return (function () {
         }
 
         function shouldPush(elt) {
-            return getClosestAttributeValue(elt, "hx-push-url") === "true" ||
+            var pushUrl = getClosestAttributeValue(elt, "hx-push-url");
+            return (pushUrl && pushUrl !== "false") ||
                 (elt.tagName === "A" && getInternalData(elt).boosted);
         }
 
+        function getPushUrl(elt) {
+            var pushUrl = getClosestAttributeValue(elt, "hx-push-url");
+            return (pushUrl === "true" || pushUrl === "false") ? "" : pushUrl;
```

Comment:
could you return null rather than an empty string here?  I don't like taking advantage of the empty string coercion, null feels more correct to me.

Reply:
Done.